operator/controllers/lifecycle/keptnworkloadinstance/controller.go

```diff
@@ -305,9 +305,14 @@ func (r *KeptnWorkloadInstanceReconciler) getAppVersionForWorkloadInstance(ctx c
 		return false, latestVersion, err
 	}
 
+	// If the latest version is empty or the workload is not found, return false and empty result
 	if latestVersion.Spec.Version == "" || !workloadFound {
 		return false, klcv1alpha3.KeptnAppVersion{}, nil
 	}
+	// If the latest version's name is longer than the maximum allowed length, truncate it
```

Comment:
i think we can remove this check here after all, as we do the check for a match on the spec.AppName property of the AppVersions, as opposed to the name of the AppVersion object